httpclient.go

```diff
@@ -25,9 +32,60 @@ func DefaultHttpClient() *http.Client {
 		ExpectContinueTimeout: 1 * time.Second,
 	}
 
-	httpClient := &http.Client{
+	return tr, nil
+}
+
+func newHttpClient(tr *http.Transport) *http.Client {
+	return &http.Client{
 		Transport: tr,
 		Timeout:   DefaultHttpTimeout,
 	}
-	return httpClient
+}
+
+// DefaultHttpClient will set up a basic client
+// with default timeouts/proxies/etc.
+func DefaultHttpClient() *http.Client {
+	tr, err := newHttpTransport("")
+	if err != nil {
+		log.Fatal().Err(err).Msg("unexpectedly failed to create a default HttpClient")
+	}
+
+	return newHttpClient(tr)
+}
+
+type HttpClientOpts struct {
+	// Proxy is the string representation of the proxy the client
+	// should use for connections.
+	Proxy string
+}
+
+func HttpClient(opts *HttpClientOpts) (*http.Client, error) {
+	tr, err := newHttpTransport(opts.Proxy)
+	if err != nil {
+		return nil, err
+	}
+
+	return newHttpClient(tr), nil
+}
+
+type proxyFunc func(*http.Request) (*url.URL, error)
+
+// newProxyFunc will use a standard ProxyFromEnvironment if no proxy
+// information is provided. Otherwise, parse the proxy string, and use
+// it as the proxy endpoint.
+func newProxyFunc(proxy string) (proxyFunc, error) {
+	proxyFn := http.ProxyFromEnvironment
+
+	if proxy != "" {
+		proxyURL, err := url.Parse(proxy)
+		if err != nil {
+			return nil, err
+		}
+
+		proxyFn = func(*http.Request) (*url.URL, error) {
```

Comment:
why not use `https://pkg.go.dev/net/http#ProxyURL`

Reply:
Because I didn't know about this :) .